New Doom Larry Enemy and Improvements to Flicker Animation for Ordinary Larries...

datahead8888 opened the pull request:
This changes introduces the following:
* Doom Larry is a usable enemy (using graphics by @Bugsbane)
* Larry has improved fuse animations (using graphics by @Bugsbane)
* Doom Larry kills Alex in one hit, while ordinary Larry only downgrades Alex.  Doom Larry is worth more points than ordinary Larry.
* Doom Larry has been given an a special sound effect for explosions.
* Script support for spawning Doom Larries is added.  This is consistent with the methods written for ordinary Larries.

Test Level: http://forum.secretchronicles.de/topics/313

As far as code reviews go, one possible change would be to try and find a way to use functionality from the cLarry class's Update method for the inherited cDoomLarry class's Update method.  The TSCError statement is, however, slightly different between the two, and I think it's a bit handy to have some sort of Update implementation in the cDoomLarry class.  I generally tried to add only methods that absolutely had to be added in the inherited cDoomLarry class.

I had intended to use the NewLarries branch, but I was apparently forced to create a NewLarries2 branch when the rebased version failed to push to it in github.  These were the relevant steps I used and the error message:
~~~~~~~~~~~~~~~~~~~~~~~~~
#Get Devel up to date locally
git checkout devel
git pull origin devel
#Do the rebase from devel to NewLarries
git checkout NewLarries
git rebase devel
#Now try to push the rebased copy to github
git push https://github.com/Secretchronicles/TSC NewLarries

Username for 'https://github.com': datahead8888
Password for 'https://datahead8888@github.com': 
To https://github.com/Secretchronicles/TSC
 ! [rejected]        NewLarries -> NewLarries (non-fast-forward)
error: failed to push some refs to 'https://github.com/Secretchronicles/TSC'
To prevent you from losing history, non-fast-forward updates were rejected
Merge the remote changes (e.g. 'git pull') before pushing again.  See the
'Note about fast-forwards' section of 'git push --help' for details.
~~~~~~~~~~~~~~~~~~~~~~~~~

At that point I created the NewLarries2 branch and pushed that to github instead.  Does anyone know why my NewLarries branch was rejected?  I have not seen anyone else having to create alternate branches for rebases.

Quintus commented on tsc/src/enemies/doom_larry.cpp:
```diff
@@ -0,0 +1,168 @@
+void cDoomLarry::Init()
+{
+    //The Doom_Larry constructor will have called the Larry constructor, creating an image set full of grey Larry images that we do not want
+    //Remove them so that they can be replaced with Doom Larry images.  If Init() is called again, this clear will also be needed again.
```
Please wrap your comments at around 70-80 chars, it gets hard to read otherwise (scrolling right).

Quintus commented:
You are missing an entry for DoomLarry in `docs/scripting/index.html` so he is linked somewhere.

Other than that, the code looks fine. I will test it later.

Vale,
Quintus

Quintus commented:
I now played and thus tested this. Looks fine to me. @datahead8888, once you fixed the two things I pointed out this can be merged.

Vale,
Quintus

datahead8888 commented:
These changes are made.  I also added an entry for Larry in index.md (it was missing, too) and fixed a minor typo in the doc.

However, it looks like Doom Larries may actually be yielding 0 points when destroyed.  I should probably play with it a little more and see what I make of it.

Quintus commented:
datahead8888 <notifications@github.com> writes:

> However, it looks like Doom Larries may actually be yielding 0 points
> when destroyed.  I should probably play with it a little more and see
> what I make of it.

The variable responsible for that is “m_kill_points”. You might want to
check whether it gets overwritten by an Init() or so and reset to zero.

Vale,
Quintus

-- 
#!/sbin/quintus
Blog: http://www.guelkerdev.de

GnuPG key: F1D8799FBCC8BC4F

datahead8888 commented:
The scoring issues are now fixed for both Larry and Doom Larry.  Please re-review this PR for merging into devel.  The NewLarries2 branch is also up to date with devel again.

Regarding the change log that was introduced, that was not a requirement at the time this PR was first created.  I like the idea of putting this change in the change log, but I would rather it be a separate commit in devel after this is merged.  We also don't know the date for the change log for certain until it is merged into devel.  Lastly, I think we should discuss the change log more with the team in a forum discussion in the mailing list, if we intend to add it as another code review requirement.

datahead8888 commented:
The Doom Larry test level is also improved: http://forum.secretchronicles.de/topics/313

Quintus commented:
I reviewed the code and tested the level. Looks fine to me, please merge! And btw. these are a *lot* of larries. What a chain explosion!

> Regarding the change log that was introduced, that was not a requirement at the time this PR was first created. I like the idea of putting this change in the change log, but I would rather it be a separate commit in devel after this is merged. 

Yep. Just merge. ChangeLog entries should be added by the people who do the merge, not the people who file the PR.

Vale,
Quintus

datahead8888 commented:
This is now merged and the change log updated
